docs/developer-docs/getting-started/write-smart-contracts.mdx

```diff
@@ -0,0 +1,624 @@
+        if (randomBytes.size() > 0) {
+            // Use the first 2 bytes to generate a number between 0 and 255
+            let bytes : [Nat8] = Blob.toArray(randomBytes);
+            currentRandomNumber := Nat8.toNat(bytes[0]);
+            Debug.print("Generated new random number: " # Nat.toText(currentRandomNumber));
+        };
+    };
+
+    // Use a query call to get the current random number.
+    public query func getCurrentNumber() : async Nat {
+        currentRandomNumber
+    };
+
+    // Initialize a timer to generate a new number every 5 seconds.
+    let timer = Timer.recurringTimer<system>(#seconds 5, generateNewNumber);
+
+    // highlight-start
+
+    // Define a public function to get the daily stats about ICP from the ICP API.
+    public func getIcpInfo() : async Text {
+
+      // This code provides the logic to call the http_request method of the management canister.
+      let HttpRequest : actor {
+        http_request(Types.HttpRequestArgs) : async Types.HttpResponsePayload
+      } = actor "aaaaa-aa";
+
+      // Define the API endpoints to obtain data from.
+      // This example uses the IC API, but any API endpoint can be used.
+      let url = "https://ic-api.internetcomputer.org/api/v3/daily-stats?format=json";
+      let transform_context : Types.TransformContext = {
+        function = transform;
+        context = Blob.fromArray([]);
+      };
+
+      // Define the http_request components.
+      let http_request : Types.HttpRequestArgs = {
+          url = url;
+          max_response_bytes = null; //optional for request
+          headers = [];
+          body = null; //optional for request
+          method = #get;
+          transform = ?transform_context;
+      };
+
+      // HTTP outcalls require cycles to be attached to the call.
+      Cycles.add<system>(20_949_972_000);
+
+      // Define the http_response
+      let http_response : Types.HttpResponsePayload = await HttpRequest.http_request(http_request);
+
+      let response_body: Blob = Blob.fromArray(http_response.body);
+      switch (Text.decodeUtf8(response_body)) {
+          case null { "No value returned" };
+          case (?y) { y };
+      };
+    };
+
+    // Define a transform function to return the status response and body.
+    public query func transform(raw : Types.TransformArgs) : async Types.CanisterHttpResponsePayload {
+        {
+            status = raw.response.status;
+            body = raw.response.body;
+            headers = [ ];
+        };
+    };
+// highlight-end
+}
+```
+
+This HTTP outcalls code references custom types, as indicated by the import statement `import Types "Types";` and code statements like `Types.CanisterHttpResponsePayload`. Custom type definitions can be created from within the `main.mo` file, but for types that may be used in several different source code files, it can be beneficial to define them in a single `Types.mo` file, and then import them as needed.
```

Comment:
we can get rid of a lot of (unnecessary) type definitions if we adopt to the concept mentioned here: https://github.com/dfinity/portal/pull/3768/files#r1846541430

for the icp.ninja example back then I started exploring this in following PR: https://github.com/marc0olo/icp-daily-planner-example/pull/1/files